i18n/detect.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python
+"""
+Function to check whether or not the English strings are up to date
+"""
+from __future__ import print_function
+from subprocess import CalledProcessError
+
+from i18n import Runner
+from i18n.execute import execute
+from i18n.extract import Extract
+
+
+class Detect(Runner):
+    """
+    Class used to check if the English source strings are up to date
+    """
+    def run(self, args):
+        """
+        Main entry point of script
+        """
+        Extract().run(args)  # Run extract command
+        changes = self.detect_changes()
+        message = self.get_message(changes)
+        print(message)
+
```

Comment:
That is exactly what I mean. The primary goal of this work is to ensure that developers do not commit new code without also submitting changed source strings. Given that all code goes through Travis, failing Travis builds is the best way to ensure developers take notice of missing data.